src/routes/students/Profile.svelte

```diff
@@ -1 +1,88 @@
+<script>
+  import {getProfile, getEmails, addEmail, setPrimaryEmail, deleteEmail} from '../../api'
+
+  const profilePromise = getProfile();
+
+  let emailsPromise = getEmails();
+
+  let newEmailValue = '';
+
+  let addEmailOutput = '';
+  const addEmailButton = () => {
+    addEmail(newEmailValue).then(res => {
+      if(res['ok'] !== undefined && !res.ok) {
+        res.json().then(output => {
+          addEmailOutput = output['email'][0];
+        });
+      } else {
+        emailsPromise = getEmails();
+        newEmailValue = '';
+        addEmailOutput = '';
+      }
+    });
+  };
+
+  const setPrimaryEmailButton = (emailId) => {
+    setPrimaryEmail(emailId).then(res => {
+      emailsPromise = getEmails();
+    });
+  };
+
+  const removeEmail = (emailId) => {
+    deleteEmail(emailId).then(res => {
+      emailsPromise = getEmails();
+    });
+  }
+</script>
+
+<style>
+  .email{
+    border: black solid 1px;
+    margin-bottom: 10px;
+  }
+</style>
+
 <div>Profile</div>
+
+<div>
+  {#await profilePromise}
+    <p>...loading profile</p>
+  {:then profile}
+    {profile['first_name']}<br>
+    {profile['last_name']}<br>
+    {profile['email']}<br>
```

Comment:
This still shows the old primary e-mail after changing it to a new one

Reply:
Could you elaborate? I cannot reproduce this bug.

Reply:
Re-requesting profile after primary email change.